### PR TITLE
Sonar cloud issues 3

### PR DESCRIPTION
**Pull request by prasa7**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApiProductsApiServiceImpl.java [hunk 1]
```diff
@@ -110,14 +110,20 @@ public class ApiProductsApiServiceImpl implements ApiProductsApiService {
...
             }
 
-            boolean isAPIPublishedOrDeprecated = APIStatus.PUBLISHED.getStatus().equals(apiProduct.getState()) ||
-                    APIStatus.DEPRECATED.getStatus().equals(apiProduct.getState());
+            String productState;
+            if (apiProduct != null) {
+                productState = apiProduct.getState();
+                apiProduct.setOrganization(organization);
+            } else {
+                productState = null;
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApiProductsApiServiceImpl.java:118]: no need else

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApiProductsApiServiceImpl.java [hunk 2]
```diff
@@ -110,14 +110,20 @@ public class ApiProductsApiServiceImpl implements ApiProductsApiService {
                 RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_API_PRODUCT, apiProductId, log);
             }
 
-            boolean isAPIPublishedOrDeprecated = APIStatus.PUBLISHED.getStatus().equals(apiProduct.getState()) ||
-                    APIStatus.DEPRECATED.getStatus().equals(apiProduct.getState());
+            String productState;
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApiProductsApiServiceImpl.java:113]: String productState = null;

**Review by tharindu1st — CHANGES_REQUESTED** [review]

**Comment by chamilaadhi** [comment]
@prasa7 @tharindu1st shall we get this merged?

**Comment by prasa7** [comment]
Closing the PR until the PR conflicts are fixed.

**Pull request closed without merging**